From e30e9c45f7387baa18a723e69afa798a3b119f19 Mon Sep 17 00:00:00 2001
From: Vidhya Vidhyaraman <vraman@marvell.com>
Date: Thu, 3 Sep 2020 17:53:03 -0400
Subject: [PATCH 15/15] octeontx2: Add IPv6 fields to default MKEX profile

commit 765264401673131f94bee0c52801728d688b01b5 from
git@git.assembla.com:cavium/WindRiver.linux.git

Added some IPv6 protocol fields to the default MKEX profile.
They include everything from the beginning of IP header and up to
source address. The pattern occupies full KW2 in MCAM entry.
Only one out of two LD registers for this protocol is used.

Signed-off-by: Vidhya Vidhyaraman <vraman@marvell.com>
Change-Id: I981847a5741b35b9218f62c1af2cd0a1319ec60d
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/35328
Reviewed-by: Yuri Tolstov <Yuri.Tolstov@cavium.com>
Reviewed-by: Devapraba Muthumani <dmuthumani@marvell.com>
Tested-by: Devapraba Muthumani <dmuthumani@marvell.com>
(cherry picked from commit cb6f43f99a37cd0d92fff31f1efe1dc96f21710b)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/35390
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index c26375d837ce..8a5d5a92bda0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -14071,6 +14071,11 @@ static struct npc_mcam_kex npc_mkex_default = {
 				/* TOS: 1 byte, KW1[63:56] */
 				KEX_LD_CFG(0x0, 0x1, 0x1, 0x0, 0xf),
 			},
+			/* Layer C: IPv6 */
+			[NPC_LT_LC_IP6] = {
+				/* Everything up to SADDR: 8 bytes, KW2[63:0] */
+				KEX_LD_CFG(0x07, 0x0, 0x1, 0x0, 0x10),
+			},
 		},
 		[NPC_LID_LD] = {
 			/* Layer D:UDP */
-- 
2.17.1

